src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -632,8 +638,100 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
             {
                 await TryHandleTimeoutAsync(invokeTask, CancellationToken.None, throwOnTimeout, timeoutTokenSource.Token, timerInterval, instance, null);
             }
+            
+            // Get the list of filters to invoke if there are any
+            List<IFunctionInvocationFilter> filters = new List<IFunctionInvocationFilter>();
+            FunctionResult functionResult = null;
+            filters = GetFilters(instance, invoker);
+
+            try
+            {
+                await InvokeExecutingFilters(filters, parameterNames, invokeParameters, functionCancellationTokenSource, instance, logger, jobHost, config);
```

Comment:
As Mike had pointed out in a previous comment, we need all of this filter stuff to be within the Timeout logic.